app/src/main/java/com/github/se/polyfit/ui/screen/IngredientScreen.kt

```diff
@@ -0,0 +1,214 @@
+) {
+
+  val initialSuggestions = 3
+  val potentialIndex = remember {
+    // equivalent to min(3, potentialIngredients.size)
+    mutableIntStateOf(initialSuggestions.coerceAtMost(potentialIngredients.size))
+  }
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(it)
+              .verticalScroll(rememberScrollState())
+              .testTag("IngredientsList"),
+      verticalArrangement = Arrangement.Top,
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        if (ingredients.isEmpty() && potentialIngredients.isEmpty()) {
+          Text(
+              "No ingredients added yet",
+              modifier = Modifier.fillMaxSize().padding(16.dp, 0.dp).testTag("NoIngredients"),
+              color = MaterialTheme.colorScheme.secondary,
+              fontSize = MaterialTheme.typography.headlineSmall.fontSize)
+        } else {
+          FlowRow(
+              horizontalArrangement = Arrangement.Start,
+              verticalArrangement = Arrangement.Top,
+              maxItemsInEachRow = Int.MAX_VALUE) {
+                ingredients.forEach { ingredient ->
+                  GradientButton(
+                      text = "${ingredient.name} ${ingredient.quantity}${ingredient.unit}",
+                      onClick = {}, // TODO: Expand to see more information
+                      active = true,
+                      modifier = Modifier.testTag("Ingredient"))
+                }
+                potentialIngredients.take(potentialIndex.intValue).forEachIndexed {
+                    index,
+                    ingredient ->
+                  GradientButton(
+                      icon = {
+                        Icon(
+                            imageVector = Icons.Default.Add,
+                            contentDescription = "Add ${ingredient.name}",
+                            tint = MaterialTheme.colorScheme.secondary)
+                      },
+                      text = ingredient.name,
+                      onClick = { onAddIngredient(index) },
+                      active = false,
+                      modifier = Modifier.testTag("PotentialIngredient"))
+                }
+                if (potentialIngredients.size > potentialIndex.intValue) {
+                  GradientButton(
+                      icon = {
+                        Icon(
+                            imageVector = Icons.Default.MoreVert,
+                            contentDescription = "More Options",
+                            modifier = Modifier.graphicsLayer(rotationZ = 90f),
+                            tint = MaterialTheme.colorScheme.secondary)
+                      },
+                      onClick = { potentialIndex.intValue = potentialIngredients.size },
+                      active = false,
+                      modifier = Modifier.testTag("MoreIngredientsButton"))
+                }
+              }
+        }
+      }
+}
+
+@Composable
+fun TopBar(navigation: Navigation) {
```

Comment:
I noticed you've developed a custom component in Jetpack Compose; however, similar functionality is already available in the existing [Compose App Bars](https://developer.android.com/develop/ui/compose/components/app-bars). Just a heads up for next time.

Reply:
Oh that is cool I didn't know, ill update to use this to make things consistent